File: src/main/java/icu/samnyan/aqua/sega/maimai2/dao/userdata/UserGeneralDataRepository.java
package icu.samnyan.aqua.sega.maimai2.dao.userdata;

import icu.samnyan.aqua.sega.maimai2.model.userdata.UserDetail;
import icu.samnyan.aqua.sega.maimai2.model.userdata.UserGeneralData;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Repository("Maimai2UserGeneralDataRepository")
public interface UserGeneralDataRepository extends JpaRepository<UserGeneralData, Long> {

    List<UserGeneralData> findByUser_Card_ExtId(long userId);

    Optional<UserGeneralData> findByUserAndPropertyKey(UserDetail user, String key);

    Optional<UserGeneralData> findByUser_Card_ExtIdAndPropertyKey(long userId, String key);

    @Transactional
    void deleteByUser(UserDetail user);
}

File: src/main/java/icu/samnyan/aqua/sega/chusan/dao/userdata/UserGeneralDataRepository.kt
package icu.samnyan.aqua.sega.chusan.dao.userdata

import icu.samnyan.aqua.sega.chusan.model.userdata.UserData
import icu.samnyan.aqua.sega.chusan.model.userdata.UserGeneralData
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Repository("ChusanUserGeneralDataRepository")
interface UserGeneralDataRepository : JpaRepository<UserGeneralData, Long> {
    fun findByUserAndPropertyKey(user: UserData, key: String): Optional<UserGeneralData>

    fun findByUser_Card_ExtIdAndPropertyKey(extId: Long, key: String): Optional<UserGeneralData>

    fun findByUser_Card_ExtId(extId: Long): List<UserGeneralData>
}
